backend/algorithms/create.py

```diff
@@ -168,67 +215,42 @@ def make_condition(tokens, first=False, course=None):
         elif is_course(token):
             # Condition for a single course
             result.add_condition(CourseCondition(token))
-        elif is_uoc(token):
-            # Condition for UOC requirement
-            uoc = get_uoc(token)
-            uoc_cond = UOCCondition(uoc)
+        elif is_program(token):
+            result.add_condition(ProgramCondition(token))
+        elif is_specialisation(token):
+            result.add_condition(SpecialisationCondition(token))
+        elif is_program_type(token):
+            result.add_condition(ProgramTypeCondition(token))
+        else:
```

Comment:
??? wierd style

Reply:
are you talking about the else statement?

Reply:
yea... there is an else statement followed by if statements that look similar

Reply:
It's because those if statements in the else all shared a large chunk of code (the remainder of the else statement).
A little weird to read but avoids a lot of duplication

Reply:
oh my bad - thats ok then